runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/ValidateRunnerXlangTest.java

```diff
@@ -287,7 +287,11 @@ protected void pythonDependenciesTest(Pipeline pipeline) {
   @RunWith(JUnit4.class)
   public static class SingleInputOutputTest extends ValidateRunnerXlangTestBase {
     @Test
-    @Category({ValidatesRunner.class, UsesCrossLanguageTransforms.class})
+    @Category({
+      ValidatesRunner.class,
```

Comment:
Is there value in running both Java and Python pipelines with Java and Python expansion services ? I wonder if we can simplify (and hopefully reduce flakiness) by running Java tests only with Python expansion service and vice versa.

Reply:
Tests using the same type of expansion service with the SDK will be skipped:
https://github.com/apache/beam/blob/bbc1867e6f3bcdb426e7f95ba923428017c0f151/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L2356
https://github.com/apache/beam/blob/bbc1867e6f3bcdb426e7f95ba923428017c0f151/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L2386